### ISSUE TITLE
System Tests fail due to ClassCastException

### ISSUE THREAD

**Issue by schweikl** [VoglSebastian/evosuite#44]
Many system tests fail due to a `ClassCastException`. For example, `NSGAIISystemTest::testIntegration`:
```
java.lang.ClassCastException: org.evosuite.testsuite.TestSuiteChromosome cannot be cast to org.evosuite.ga.NSGAChromosome
	at org.evosuite.ga.metaheuristics.NSGAIISystemTest.testIntegration(NSGAIISystemTest.java:331)
```
In this particular example, the relevant lines appear to be the following ones:
https://github.com/VoglSebastian/evosuite/blob/91cad2519b06e68bc3f3a1fcfc02d105e661ac8a/master/src/test/java/org/evosuite/ga/metaheuristics/NSGAIISystemTest.java#L322-L331
Line 323 contains an unchecked cast:
```Java
(GeneticAlgorithm<NSGAChromosome>) getGAFromResult(result);
```
It is assumed that `getGAFromResult` yields a `GeneticAlgorithm<NSGAChromosome>`. But in reality, it should be a `GeneticAlgorithm<TestSuiteChromosome>`. If you change the following lines in that respect, too, the system test at least compiles and runs without mentioned `ClassCastException`. However, it will still fail due to an `AssertionError`:
```
java.lang.AssertionError: 
Expected :0.0
Actual   :0.16666666666666669
at org.evosuite.ga.metaheuristics.NSGAIISystemTest.testIntegration(NSGAIISystemTest.java:333)
```

I suspect that the `ClassCastException`s thrown by many other system tests can be fixed in a similar way. However, the `AssertionError` still requires further investigation…

**Comment by schweikl** [comment]
Update: I have just pushed a new branch [`fixSystemTestClassCastException`](https://github.com/VoglSebastian/evosuite/tree/fixSystemTestClassCastException) containing the proposed fix for the `ClassCastException` occurring in `NSGAIISystemTest::testIntegration`

**Comment by schweikl** [comment]
Update 2: the only other `ClassCastException` is caused by `SearchStatisticsTest::test_setOutputVariable`:
```
java.lang.ClassCastException: class org.evosuite.statistics.SearchStatistics$CoverageSequenceOutputVariableFactory cannot be cast to class org.evosuite.statistics.DirectSequenceOutputVariableFactory (org.evosuite.statistics.SearchStatistics$CoverageSequenceOutputVariableFactory and org.evosuite.statistics.DirectSequenceOutputVariableFactory are in unnamed module of loader 'app')
	at org.evosuite.statistics.SearchStatisticsTest.test_setOutputVariable(SearchStatisticsTest.java:10)
```
But this seems to be a separate (non-related) issue.

I will go ahead, merge `fixSystemTestClassCastException` into `selfType` and close this issue.

**Comment by schweikl** [comment]
Fixed by #45
